src/jvm/main/org/jetbrains/kotlinx/lincheck/util/Types.kt

```diff
@@ -36,25 +36,62 @@ internal fun MethodDescriptor.isArraysCopyOfIntrinsic(): Boolean {
     return (
         className == "java.util.Arrays" &&
         methodName == "copyOf" &&
-        returnType == ARRAY_OF_OBJECTS_TYPE &&
-        argumentTypes == listOf(ARRAY_OF_OBJECTS_TYPE, INT_TYPE, CLASS_TYPE)
+        (
+            (
+                returnType == ARRAY_OF_OBJECTS_TYPE &&
+                argumentTypes == listOf(ARRAY_OF_OBJECTS_TYPE, INT_TYPE)
+            ) ||
+            (
+                returnType == ARRAY_OF_OBJECTS_TYPE &&
+                argumentTypes == listOf(ARRAY_OF_OBJECTS_TYPE, INT_TYPE, CLASS_TYPE)
+            ) ||
+            ARRAY_OF_PRIMITIVE_TYPES.any {
+                returnType == it &&
+                argumentTypes == listOf(it, INT_TYPE)
+            }
+        )
```

Comment:
Could you please compress this code a bit. 
Too many almost empty lines and indentation.